consensus/reactor.go

```diff
@@ -447,12 +447,17 @@ func (conR *Reactor) broadcastHasVoteMessage(vote *types.Vote) {
 }
 
 func makeRoundStepMessage(rs *cstypes.RoundState) (nrsMsg *NewRoundStepMessage) {
+	lastCommitRound := rs.LastCommit.GetRound()
+
 	nrsMsg = &NewRoundStepMessage{
 		Height:                rs.Height,
 		Round:                 rs.Round,
 		Step:                  rs.Step,
 		SecondsSinceStartTime: int(time.Since(rs.StartTime).Seconds()),
-		LastCommitRound:       rs.LastCommit.GetRound(),
+		LastCommitRound:       lastCommitRound,
+	}
+	if rs.Height > 1 && lastCommitRound == -1 {
```

Comment:
The idea here were to understand why this is happening, not overwrite the value. Sorry if the issue description was not clear enough.

Reply:
@melekes I understand the purpose of the issue. But you're right that this is not the root cause, and not the place to fix. New fix is coming.